### ISSUE TITLE
Address editing glitch

### ISSUE THREAD

**Issue by JohnVeness** [SimpleMobileTools/Simple-Contacts#482]
If I edit an address in a contact then save the contact, two addresses now show in the contact (both the original and the edited version). If I try to edit again, only the edited version shows up, so there seems to be no way to delete the old address. I eventually managed to delete the old address in a different contacts app.

6.5.1 from F-Droid.

**Comment by tibbi** [comment]
dont you have the contact stored multiple times? At the "View details" screen, do you see multiple contact sources at the bottom? Multiple emails, or so.

**Comment by JohnVeness** [comment]
Ah yes. Interestingly, for the affected contacts (and it doesn't happen to all of them) it looks like they're stored multiple times, although both in the same source (DecSync in my case). I'm not sure how that happened, or how to fix it.

**Comment by tibbi** [comment]
well contact handling on android is a huge mess, there are many possible ways duplicates are created, it isnt your fault. You can open individual contacts by clicking on emails at that screen, maybe that way you can figure out which one has an address filled out and edit that one. Or you can also open an individual contact by clicking on the email and delete it, it will delete only the given single instance. You have to pay attention to delete the proper contact though. Some merger might be added in the app someday, Im getting there step by step.

**Comment by JohnVeness** [comment]
Thanks for the pointers. Yes, it looks like I have hundreds of these identical duplicate contacts. Might have been that I imported them twice at some point?

So I guess that explains the origin of the problem. Although still, the behaviour of this app is still rather odd when in this state: view details shows two addresses but clicking edit just shows one, with no clear (to the user) reason. I don't know if that can be improved?

**Comment by tibbi** [comment]
well the View screen merges all possibly duplicates, but the Edit shows only one of them. I dont know what would be the ideal solution yet, Im working towards it step by step. It is always better, duplicate contacts were way more annoying previously :)

**Comment by JohnVeness** [comment]
Agreed, this is certainly better than multiple contacts showing in the main list (even if that would have alerted me to the source of my problem earlier).

I'm not sure what to suggest about the editing problem. Maybe a dropdown at the top showing all the sources, so the user can pick which one to edit? The current solution of clicking directly on the source name at the bottom to edit an individual one is not very discoverable compared to clicking on the edit icon at the top, although it is very useful once you know about it.

In any case, whatever you decide, and once again thanks for this app and for making it open source. :)
